Detectors/PHOS/simulation/src/Detector.cxx

```diff
@@ -0,0 +1,1017 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <algorithm>
+#include <iomanip>
+#include <map>
+
+#include "TGeoManager.h"
+#include "TGeoVolume.h"
+#include "TVirtualMC.h"
+
+#include "FairGeoNode.h"
+#include "FairRootManager.h"
+#include "FairVolume.h"
+
+#include "PHOSBase/Geometry.h"
+#include "PHOSBase/Hit.h"
+#include "PHOSSimulation/Detector.h"
+#include "PHOSSimulation/GeometryParams.h"
+
+#include "SimulationDataFormat/Stack.h"
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/range/irange.hpp>
+
+using namespace o2::phos;
+
+ClassImp(Detector);
+
+Detector::Detector(Bool_t active)
+  : o2::Base::DetImpl<Detector>("PHS", active),
+    mHits(new std::vector<Hit>),
+    mCurrentTrackID(-1),
+    mCurrentCellID(-1),
+    mCurentSuperParent(-1),
+    mCurrentHit(nullptr)
+{
+}
+
+void Detector::Initialize()
+{
+  o2::Base::Detector::Initialize();
+  Reset();
+}
+
+void Detector::EndOfEvent()
+{
+  // Sort Hits
+  // Add duplicates if any and remove them
+  // TODO: Apply Poisson smearing of light production
+
+  auto first = mHits->begin() ;
+  auto last = mHits->end() ;
+
+  std::sort(first, last);
+  
+  first = mHits->begin() ;
+  last = mHits->end() ;
+  auto itr = first ;
+  while (++first != last) {
```

Comment:
what is this doing? Might there be a more readable alternative in stl?

Reply:
this is copy of std::unique() method with addition: adding identical Hits